taker-frontend/src/components/History.tsx

```diff
@@ -93,13 +93,19 @@ const CfdDetails = ({ cfd, connectedToMaker, displayCloseButton }: CfdDetailsPro
             isForceCloseButton={true}
         />;
 
+    let failedCfd = [StateKey.REJECTED, StateKey.SETUP_FAILED].includes(cfd.state.key);
```

Comment:
We can probably learn a lot from this as to what the `CfdState` should actually be. Likely, we are exposing too many details here.